opentelemetry/instrumentation/app/work.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"context"
+	"math/rand"
+	"time"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+const name = "work"
+
+var (
+	meter         = otel.Meter(name)
+	tracer        = otel.Tracer(name)
+	workHistogram metric.Int64Histogram
+)
+
+func init() {
+	var err error
+	workHistogram, err = meter.Int64Histogram("example.histogram",
+		metric.WithDescription("Sample histogram"),
+		metric.WithUnit("1"))
+
+	if err != nil {
+		panic(err)
+	}
+}
+
+// doWork simulates a some job being triggerred in response to an API call to the server.
+// This function computes 10 random values and records them into a histogram which can be
+// later visualized as a distribution.
+func doWork(ctx context.Context, host string) time.Duration {
+	start := time.Now()
+	hostValue := attribute.String("host.value", host)
+
+	// simulate the overall work by sleeping
+	sleepTime := time.Duration(100+rand.Intn(100)) * time.Millisecond
+	time.Sleep(sleepTime)
+
+	// wrap the random number generation in a span - to better visualize the time spent in this part
+	traceCtx, span := tracer.Start(ctx, "doWork")
+	for i := 0; i < 10; i++ {
```

Comment:
Do we need the loop here?  Or do we not get enough data in the histogram without it?

Reply:
I think the loop can be removed, we are generating enough exemplars.

Reply:
+1 I would put the span around the sleep

Reply:
Put span around sleep & removed the loop.